### PR TITLE
[terra-clinical-result] Add optional chaining operator to clinical result display and update translations

### PR DESCRIPTION
**Pull request by RayGunY**
### Summary

When trying to validate the alt text for strikethroughs I see that the clinical result site page has an error message. In the dev tools I’m seeing the message “TypeError: Cannot read properties of undefined (reading 'unit')“

I believe my issue is a small check that differs between the clinical result display and the blood pressure display files.

You can see in the clinical result blood pressure display file I’m checking the unit is valid like so: https://github.com/cerner/terra-clinical/pull/870/files#diff-167a4e0b1d0ca2ee4064ed5b2c27e92feee651e13b3b21766c1ad4e538a18a77R48 

But I in the clinical result display file I didn’t use the optional chaining operator (?.): https://github.com/cerner/terra-clinical/pull/870/files#diff-7431c9460fef4d7dc39515e8a588b9374dfdf2c11c98373005aaf3e5e0b8bd3eR91 

So I went ahead and added it, 

**What was changed:**
Add optional chaining operator to clinical result display like it is in clinical result blood pressure display.

**Why it was changed:**
From my understanding - the ?. operator is like the . chaining operator, except that instead of causing an error if a reference is nullish (null or undefined), the expression short-circuits with a return value of undefined.

### Testing

This change was tested using:

- [x] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-9067 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by RayGunY** [comment]
This behavior didn't show when running my previous branch locally. Running this branch locally we see that the behavior is what is expected, hopefully once merged to main this should fix the issue.
https://github.com/cerner/terra-clinical/assets/40574651/d8ce34d4-8da7-47ce-a704-858fbbec030e

**Comment by kolkheang** [comment]
I think it would it be good to add a jest test with this scenario of no result.

File: packages/terra-clinical-result/src/_ClinicalResultDisplay.jsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ const ClinicalResultDisplay = ({
   intl,
 }) => {
   const isValidValue = result?.value;
-  const isUnitPresent = !hideUnit && result.unit;
+  const isUnitPresent = !hideUnit && result?.unit;
```
Comment (vinaybhargavar) [inline packages/terra-clinical-result/src/_ClinicalResultDisplay.jsx:91]: Can we add a jest test for this?

Comment (RayGunY) [reply packages/terra-clinical-result/src/_ClinicalResultDisplay.jsx:91]: Sure: [a704108](https://github.com/cerner/terra-clinical/pull/873/commits/a704108d7601fe249416ff5d5ade07efd9ca0efb)

**Comment by RayGunY** [comment]
> I think it would it be good to add a jest test with this scenario of no result.

I went ahead and added that test: [a704108](https://github.com/cerner/terra-clinical/pull/873/commits/a704108d7601fe249416ff5d5ade07efd9ca0efb)

**Review by kolkheang — APPROVED** [review]

**Review by vinaybhargavar — APPROVED** [review]

**Comment by RayGunY** [comment]
@sycombs @sdadn there was an issue during validation for the strikethrough PR that was merged Friday - this change should fix it. Added a test for this case as well.

**Comment by RayGunY** [comment]
The translations for the strikethrough text just came through so I went ahead and added them to this PR. 
Jira where translations were added: https://jira.cerner.com/browse/TRANS-14531

File: packages/terra-clinical-result/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 * Added
   * Added screen-reader support for strikethroughs to convey an entered in error status.
+  * Added additonal translations for strikethrough alt text.
+
+* Fixed
+  * Fixed the clincial result display check for if the unit is valid when reading alt text for strikethroughs.
```
Comment (sycombs) [inline packages/terra-clinical-result/CHANGELOG.md:10]: Can we reword this to be more consumer-oriented rather than referring to the internal `ClinicalResultDisplay` component?

File: packages/terra-clinical-result/tests/jest/ClinicalResult.test.jsx [hunk 3]
```diff
@@ -62,6 +62,17 @@ describe('ClinicalResult', () => {
...
+  it('should render a result with no unit', () => {
+    const resultData = {
+      eventId: '111',
+      result: {
+        value: '12345.678',
+      },
+    };
+    const result = shallow(<ClinicalResult {...resultData} />);
+    expect(result).toMatchSnapshot();
+  });
```
Comment (sycombs) [inline packages/terra-clinical-result/tests/jest/ClinicalResult.test.jsx:74]: Can we add a test case where `result` is undefined?

File: packages/terra-clinical-result/translations/es-ES.json [hunk 4]
```diff
@@ -2,6 +2,7 @@
   "Terra.clinicalResult.resultError": "Error",
   "Terra.clinicalResult.statusInError": "Con errores",
   "Terra.clinicalResult.statusInErrorHiddenText": "El valor no se ha documentado como Con errores",
+  "Terra.clinicalResult.statusInErrorStrikethrough": "Resultado {strikethroughResult}escrito Con errores",
```
Comment (sycombs) [inline packages/terra-clinical-result/translations/es-ES.json:5]: Looks like these are missing a space between `{strikethroughResult}` and `escrito`. Can you reach back out to the translations team to confirm if this is correct?

**Review by sycombs — CHANGES_REQUESTED** [review]

File: packages/terra-clinical-result/tests/jest/ClinicalResult.test.jsx (hunk 3 above)
Comment (RayGunY) [reply packages/terra-clinical-result/tests/jest/ClinicalResult.test.jsx:74]: When result is undefined we don't render the visually hidden text, added a test for that here: [ffc5b1d](https://github.com/cerner/terra-clinical/pull/873/commits/ffc5b1db3c0cb80217f444379f2f2e7c5805ed56)

File: packages/terra-clinical-result/CHANGELOG.md (hunk 2 above)
Comment (RayGunY) [reply packages/terra-clinical-result/CHANGELOG.md]: Sure thing, is this better? [b7c893a](https://github.com/cerner/terra-clinical/pull/873/commits/b7c893a81023fec9b20769733b305439b79839e0)

File: packages/terra-clinical-result/translations/es-ES.json (hunk 4 above)
Comment (RayGunY) [reply packages/terra-clinical-result/translations/es-ES.json]: Really good catch, thank you for calling that out. I left a comment on the Jira but I'm pretty positive you're correct, there should be a space. I went ahead and changed those translations, but can post another comment here with confirmation when they get back to me. [5057289](https://github.com/cerner/terra-clinical/pull/873/commits/505728911b13daacb552623d14b09e4c5fa31f9b)

Comment (RayGunY) [reply packages/terra-clinical-result/translations/es-ES.json]: Translation team got back to me - there is supposed to be a space so we should be good to go now.

**Review by sycombs — APPROVED** [review]

**Review by sdadn — APPROVED** [review]
